system/Test/Mock/MockEmail.php

```diff
@@ -11,14 +12,27 @@ class MockEmail extends Email
 	 */
 	public $archive = [];
 
+	/**
+	 * Value to return from mocked send().
+	 *
+	 * @var boolean
+	 */
+	public $returnValue = true;
+
 	public function send($autoClear = true)
 	{
+		$this->archive = get_object_vars($this);
+
 		if ($autoClear)
```

Comment:
I think it need to be : 

```suggestion
              if ($autoClear && ! $this->returnValue)
```

So the `archive` property will not be cleared when `retrunValue` is true.

Reply:
`clear()` does not affect `$archive` but rather a pre-defined list of properties, so I think this is fine. Your comment did make me realize though that we should only be clearing on successful returns, so I'm giving this the same consolidation treatment as `Email` above.